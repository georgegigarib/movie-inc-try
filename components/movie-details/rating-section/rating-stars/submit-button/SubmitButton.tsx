// SubmitButton.tsx
import React from 'react';
import { StyleSheet, TouchableOpacity, ActivityIndicator } from 'react-native';
import { MaterialCommunityIcons } from '@expo/vector-icons';

interface SubmitButtonProps {
  isModified: boolean;
  submitted: boolean;
  isLoading: boolean;
  onPress: () => void;
}

const SubmitButton: React.FC<SubmitButtonProps> = ({ isModified, submitted, isLoading, onPress }) => {
  return (
    <TouchableOpacity
      style={[
        styles.button,
        {
          opacity: isModified || submitted ? 1 : 0,
          backgroundColor: submitted ? '#28a745' : '#007BFF',
        },
      ]}
      onPress={onPress}
      disabled={!isModified || submitted || isLoading}
    >
      {isLoading ? (
        <ActivityIndicator size="small" color="#FFFFFF" />
      ) : submitted ? (
        <MaterialCommunityIcons name="check" size={18} color="#FFFFFF" />
      ) : (
        <MaterialCommunityIcons name="send" size={18} color="#FFFFFF" />
      )}
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  button: {
    position: 'absolute',
    right: -35,
    bottom: 21,
    paddingVertical: 5,
    paddingHorizontal: 5,
    borderRadius: 25,
    alignItems: 'center',
    justifyContent: 'center',
  },
});

export default SubmitButton;
