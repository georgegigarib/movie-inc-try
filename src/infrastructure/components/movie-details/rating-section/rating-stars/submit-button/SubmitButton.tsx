import React from 'react';
import { StyleSheet, TouchableOpacity, ActivityIndicator } from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome';

interface SubmitButtonProps {
  isModified: boolean;
  submitted: boolean;
  isLoading: boolean;
  onPress: () => void;
}

const SubmitButton: React.FC<SubmitButtonProps> = ({
  isModified,
  submitted,
  isLoading,
  onPress,
}) => {
  return (
    <TouchableOpacity
      style={[
        styles.button,
        {
          opacity: isModified || submitted ? 1 : 0,
          backgroundColor: submitted ? '#28a745' : '#007BFF',
        },
      ]}
      onPress={onPress}
      disabled={!isModified || submitted || isLoading}
      testID="submit-button"
    >
      {isLoading ? (
        <ActivityIndicator size="small" color="#FFFFFF" testID="activity-indicator" />
      ) : submitted ? (
        <Icon name="check" size={18} color="#FFFFFF" testID="check-icon" />
      ) : (
        <Icon name="send" size={18} color="#FFFFFF" testID="send-icon" />
      )}
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  button: {
    position: 'absolute',
    right: -35,
    bottom: 21,
    paddingVertical: 5,
    paddingHorizontal: 5,
    borderRadius: 25,
    alignItems: 'center',
    justifyContent: 'center',
  },
});

export default SubmitButton;
