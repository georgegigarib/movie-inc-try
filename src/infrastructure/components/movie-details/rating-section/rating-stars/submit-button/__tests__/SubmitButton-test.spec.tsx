import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import SubmitButton from '@/src/infrastructure/components/movie-details/rating-section/rating-stars/submit-button/SubmitButton';

describe('SubmitButton', () => {
  const mockOnPress = jest.fn();

  test('renders correctly when not loading, not submitted, and modified', () => {
    const { getByTestId, toJSON } = render(
      <SubmitButton
        isModified={true}
        submitted={false}
        isLoading={false}
        onPress={mockOnPress}
      />
    );
    expect(toJSON()).toMatchSnapshot();
    const button = getByTestId('submit-button');
    expect(button).toBeTruthy();
    expect(getByTestId('send-icon')).toBeTruthy();
  });

  test('renders correctly when loading', () => {
    const { getByTestId, toJSON } = render(
      <SubmitButton
        isModified={true}
        submitted={false}
        isLoading={true}
        onPress={mockOnPress}
      />
    );
    expect(toJSON()).toMatchSnapshot();
    const button = getByTestId('submit-button');
    expect(button).toBeTruthy();
    expect(getByTestId('activity-indicator')).toBeTruthy();
  });

  test('renders correctly when submitted', () => {
    const { getByTestId, toJSON } = render(
      <SubmitButton
        isModified={true}
        submitted={true}
        isLoading={false}
        onPress={mockOnPress}
      />
    );
    expect(toJSON()).toMatchSnapshot();
    const button = getByTestId('submit-button');
    expect(button).toBeTruthy();
    expect(getByTestId('check-icon')).toBeTruthy();
  });

  test('does not call onPress when disabled', () => {
    const { getByTestId } = render(
      <SubmitButton
        isModified={false}
        submitted={true}
        isLoading={false}
        onPress={mockOnPress}
      />
    );
    const button = getByTestId('submit-button');
    fireEvent.press(button);
    expect(mockOnPress).not.toHaveBeenCalled();
  });

  test('calls onPress when enabled and pressed', () => {
    const { getByTestId } = render(
      <SubmitButton
        isModified={true}
        submitted={false}
        isLoading={false}
        onPress={mockOnPress}
      />
    );
    const button = getByTestId('submit-button');
    fireEvent.press(button);
    expect(mockOnPress).toHaveBeenCalled();
  });
});
