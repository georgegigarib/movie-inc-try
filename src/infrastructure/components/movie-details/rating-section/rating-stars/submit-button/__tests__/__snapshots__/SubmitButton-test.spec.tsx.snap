// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmitButton renders correctly when loading 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#007BFF",
      "borderRadius": 25,
      "bottom": 21,
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 5,
      "paddingVertical": 5,
      "position": "absolute",
      "right": -35,
    }
  }
  testID="submit-button"
>
  <ActivityIndicator
    color="#FFFFFF"
    size="small"
    testID="activity-indicator"
  />
</View>
`;

exports[`SubmitButton renders correctly when not loading, not submitted, and modified 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#007BFF",
      "borderRadius": 25,
      "bottom": 21,
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 5,
      "paddingVertical": 5,
      "position": "absolute",
      "right": -35,
    }
  }
  testID="submit-button"
>
  <Text
    style={
      {
        "color": "#FFFFFF",
        "fontSize": 18,
      }
    }
    testID="send-icon"
  >
    send
  </Text>
</View>
`;

exports[`SubmitButton renders correctly when submitted 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#28a745",
      "borderRadius": 25,
      "bottom": 21,
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 5,
      "paddingVertical": 5,
      "position": "absolute",
      "right": -35,
    }
  }
  testID="submit-button"
>
  <Text
    style={
      {
        "color": "#FFFFFF",
        "fontSize": 18,
      }
    }
    testID="check-icon"
  >
    check
  </Text>
</View>
`;
